app/controllers/conference_registrations_controller.rb

```diff
@@ -6,22 +6,19 @@ class ConferenceRegistrationsController < ApplicationController
 
   def new
     @registration = Registration.new(conference_id: @conference.id)
-    authorize! :new, @registration, message: "Sorry, you can not register for #{@conference.title}. Registration limit exceeded or the registration is not open."
 
     # Redirect to registration edit when user is already registered
     if @conference.user_registered?(current_user)
+      authorize! :edit, current_user.registrations.find_by(conference_id: @conference.id)
       redirect_to edit_conference_conference_registration_path(@conference.short_title)
       return
-    # ichain does not allow us to create users during registration
-    elsif (ENV['OSEM_ICHAIN_ENABLED'] == 'true') && !current_user
-      redirect_to root_path, alert: 'You need to sign in or sign up before continuing.'
-      return
     end
 
-    # avoid openid sign_in to redirect to register/new when the sign_in user had already a registration
-    if current_user && @conference.user_registered?(current_user)
-      redirect_to edit_conference_conference_registration_path(@conference.short_title)
+    unless @conference.registration_open? && !@conference.registration_limit_exceeded?
```

Comment:
There are cases where `can? :new, ,@registration` will return false while the required result is true, for example if OSEM_ICHAIN_ENABLED is true, the user is not signed in, the registrations are open and the registration limit hasn't been exceeded. Therefore it isn't a suitable replacement.